src/create_maps.cpp

```diff
@@ -0,0 +1,366 @@
+    group_poses = H5Gcreate(group_reachability,"/Reachability/Poses",H5P_DEFAULT, H5P_DEFAULT, H5P_DEFAULT);
+    group_spheres = H5Gcreate(group_reachability,"/Reachability/Spheres",H5P_DEFAULT, H5P_DEFAULT, H5P_DEFAULT);
+
+
+ 
+//Creating Poses dataset: This process is little complicated. As the resolution decreses, e.g 0.03, the number of poses rises upto 7 digits, so the matrix of double value is not going to support this. We are creating a chunked infinite dataset which will select memory hyperslab and write data to it from buffer. We are using 2 hyperslabs.
+
+    ROS_INFO("Saving poses in reachability map");
+    
+    
+    const hsize_t ndims = 2;
+    const hsize_t ncols = 7;
+    
+    int posSize=PoseColFilter.size();
+    int chunk_size;
+    int PY =7;
+    if (posSize%2){
+        chunk_size=(posSize/2)+1;
+        	}
+    else{
+         chunk_size=(posSize/2);
+    }
+    
+    
+//Create Dataspace 
+    hsize_t dims[ndims] = {0, ncols};			//Starting with an empty buffer
+    hsize_t max_dims[ndims] = {H5S_UNLIMITED, ncols};			//Creating dataspace
+    hid_t file_space = H5Screate_simple(ndims, dims, max_dims);
+    
+//Create Dataset Property list 
+    hid_t plist = H5Pcreate(H5P_DATASET_CREATE);
+    H5Pset_layout(plist, H5D_CHUNKED);
+    hsize_t chunk_dims[ndims] = {chunk_size, ncols};
+    H5Pset_chunk(plist, ndims, chunk_dims);
+
+//Create the datset
+    hid_t dset = H5Dcreate(group_poses, "poses_dataset", H5T_NATIVE_FLOAT, file_space, H5P_DEFAULT, plist, H5P_DEFAULT);
+    
+
+//Closing resources
+    H5Pclose(plist);
+    H5Sclose(file_space);
+
+//Creating the first buffer
+    hsize_t nlines = chunk_size;
+    float *buffer = new float[nlines * ncols];
+    float **dset1_data = new float*[nlines];
+    for (hsize_t i = 0; i < nlines; ++i){
+        dset1_data[i] = &buffer[i * ncols];
+    }
+    
+//Data for the first chunk     
+    for (int i=0;i<chunk_size;i++){
+	for(int j=0;j<PY;j++){
+	   dset1_data[i][j]=poseReach[i][j];
+	   }
+     }
+    
+//Memory dataspace indicating size of the buffer    
+    dims[0] = chunk_size;
+    dims[1] = ncols;
+    hid_t mem_space = H5Screate_simple(ndims, dims, NULL);
+    
+
+//Extending dataset
+    dims[0] = chunk_size;
+    dims[1] = ncols;
+    H5Dset_extent(dset, dims);
+    
+
+//Selecting hyperslab on the dataset
+    file_space = H5Dget_space(dset);
+    hsize_t start[2] = {0, 0};
+    hsize_t count[2] = {chunk_size, ncols};
+    H5Sselect_hyperslab(file_space, H5S_SELECT_SET, start, NULL, count, NULL);
+    
+//Writing buffer to the dataset
+    H5Dwrite(dset, H5T_NATIVE_FLOAT, mem_space, file_space, H5P_DEFAULT, buffer);
+    
+
+//Closing file dataspace
+    H5Sclose(file_space);
+
+//Data for the Second chunk	
+    for (int i=chunk_size;i<posSize;i++){
+	for(int j=0;j<PY;j++){
+	   
+	   dset1_data[i-chunk_size][j]=poseReach[i][j];
+	   }
+     } 
+    
+   
+//Resizing new memory dataspace indicating new size of the buffer
```

Comment:
@akgoins The lines 239-258 is creating the first buffer and selecting the hyperslabs on the dataset, the lines 273-291 is doing almost the same but for the next chunk of data. The selection of limit of the size of the dataset was result of rigorous testing.  :)  As the h5 in c++ with the hyperslabs is pretty complex, I would not recommend doing it in a private function right now. The file is still going to change in the upcoming PRs. I will surely try to make it a function when the file is final.